Completed Event Header for view page

Clotonervo opened the pull request:
Let me know what changes need to go into this one 👍 💯 🥇 🔢 💯 👍 🥇 🥇 💯 👍 🔢 💯 💯

Commit:
Completed Event Header for view page

hokansom commented on client/src/components/EventView/Header/index.js:
```diff
@@ -0,0 +1,128 @@
+  onPressSettings: PropTypes.func,
+
+  /** A callback rasied when the title is edited. */
+  onEditTitle: PropTypes.func
```
This component doesn't need to edit the title, so you can remove this prop

hokansom commented on client/src/components/EventView/Header/index.js:
```diff
@@ -0,0 +1,128 @@
+            <TextContainer>{title}</TextContainer>
+        </TitleContainer>
+        <DateContainer>
+          <DateInput
```
We don't want the date to be editable, so you would want to make your own component to just display the date

hokansom commented on client/src/components/EventView/Header/index.js:
```diff
@@ -0,0 +1,128 @@
+`;
+
+const TitleContainer = styled.div`
+  width: 60%;
```
For this component, we wouldn't want to set a width. If the event has a super long title, we would want it to extend to the other margin, and then wrap if necessary.

Commit:
Fixed some issues mentioned before

Commit:
Clean up some comments and other things

Clotonervo commented:
I believe I fixed those things you mentioned before. I'm pretty rusty with React, so let me know if I did something wrong there haha. The one thing that I wasn't sure was ok is that I changed the `width: 60%` to be just `width: 100%` as I figured that that would just be easier than restructuring the rest of the html structure. If that needs to be changed let me know. Also I had a hard time getting the date to display correctly, so I ended up just using a function on the date object that converted it into a string. I didn't know if this mattered but it wasn't as simple as last time we were using a date picker to display the date rather than doing it ourselves. Let me know what you think!

hokansom commented on client/src/components/EventView/Header/index.js:
```diff
@@ -0,0 +1,116 @@
+            <TextContainer>{title}</TextContainer>
+        </TitleContainer>
+        <DateContainer>
+          <DateTextContainer>{realDate.toDateString()}</DateTextContainer>
```
Josiah created a useful component for formatting dates. It's located in /Common/Date. You can add a new DateFormat to it with whatever format you want, like "MMMM D, YYYY", which would print out December 3, 2020. Let me know if you need more help with that part

hokansom reviewed (changes requested):
Everything looks good,  but we can improve the date display. See my other comment.

Commit:
Changes to Date for formatting

hokansom committed:
add new date format

hokansom reviewed (approved):
Looks good!
